mods/ra/maps/sarin-gas-1-crackdown/crackdown.lua

```diff
@@ -124,25 +100,35 @@ SendPatrol = function(mammoth)
 end
 
 MoveTruckNorth = function(truck)
+	MoveTruckEscapeRoute(truck, TruckEscapeNorth)
+end
+
+MoveTruckSouth = function(truck)
+	MoveTruckEscapeRoute(truck, TruckEscapeSouth)
+end
+
+MoveTruckEscapeRoute = function(truck, route)
 	if truck.IsDead then
 		return
 	else
 		Media.DisplayMessage("Convoy truck attempting to escape!")
 		Media.PlaySoundNotification(greece, "AlertBleep")
-		Utils.Do(TruckEscapeNorth, function(waypoint)
+		Utils.Do(route, function(waypoint)
 			truck.Move(waypoint.Location)
 		end)
+		SetTruckEscapeTrigger(truck, route[#route])
 	end
 end
 
-MoveTruckSouth = function(truck)
-	if truck.IsDead then
-		return
-	else
-		Media.DisplayMessage("Convoy truck attempting to escape!")
-		Media.PlaySoundNotification(greece, "AlertBleep")
-		Utils.Do(TruckEscapeSouth, function(waypoint)
-			truck.Move(waypoint.Location)
+SetTruckEscapeTrigger = function (actor, escape)
+	if actor.Owner == ussr and actor.Type == "truk" then
```

Comment:
We don't need this check any more. At the time this function is running we know that `actor` is a truck and that it is owned by the soviets.

Reply:
right